amplipi/streams/spotify_connect.py

```diff
@@ -0,0 +1,198 @@
+    player_args = f"{utils.get_folder('streams')}/go-librespot --config_dir {src_config_folder}".split(' ')
+    logger.debug(f'spotify player args: {player_args}')
+
+    self.proc = subprocess.Popen(args=player_args, stdin=subprocess.PIPE,
+                                 stdout=self._log_file, stderr=self._log_file)
+
+    url = f"http://localhost:{self._api_port}"
+    meta_reader = f"{utils.get_folder('streams')}/spot_connect_meta.py"
+    meta_args = [sys.executable, meta_reader, url, self.meta_file, '--debug']  # TODO: remove --debug for production
+    logger.info(f'{self.name}: starting metadata reader: {meta_args}')
+    self.proc2 = subprocess.Popen(args=meta_args, stdout=self._log_file, stderr=self._log_file)
+
+  def _deactivate(self):
+    if self._is_running():
+      self.proc.stdin.close()
+      logger.info(f'{self.name}: stopping player')
+      self.proc.terminate()
+      self.proc2.terminate()
+      if self.proc.wait(1) != 0:
+        logger.info(f'{self.name}: killing player')
+        self.proc.kill()
+      if self.proc2.wait(1) != 0:
+        logger.info(f'{self.name}: killing metadata reader')
+        self.proc2.kill()
+      self.proc.communicate()
+      self.proc2.communicate()
+    if self.proc and self._log_file:  # prevent checking _log_file when it may not exist, thanks validation!
+      self._log_file.close()
+    if self.src:
+      try:
+        subprocess.run(f'rm -r {utils.get_folder("config")}/srcs/{self.src}/*', shell=True, check=True)
+      except Exception as e:
+        logger.exception(f'{self.name}: Error removing config files: {e}')
+    self._disconnect()
+    self.proc = None
+    self.proc2 = None
+
+  def info(self) -> models.SourceInfo:
+    source = models.SourceInfo(
+      name=f"Connect to {self.name} on Spotify Connect",
+      state=self.state,
+      img_url='static/imgs/spotify.png',
+      type=self.stream_type
+    )
+
+    if not self.meta_file:
+      logger.error(f'{self.name}: no metadata file. info() called on un-activated stream')
+      return source
+
+    if not os.path.exists(self.meta_file):
+      return source
+
+    try:
+      with open(self.meta_file, 'r', encoding='utf8') as f:
+        metadata = yaml.safe_load(f)
+      if 'track' not in metadata or not metadata['track']['name']:
+        return source  # no track info, there is probably no device connected
+      source.name = self.full_name()
+      source.track = metadata['track']['name']
+      source.album = metadata['track']['album_name']
+      source.artist = ", ".join(metadata['track']['artist_names'])
+      if metadata['stopped']:
+        source.state = "stopped"
+        source.supported_cmds = ['play']
+      elif metadata['paused']:
+        source.state = "paused"
+        source.supported_cmds = self.supported_cmds
+      else:
+        source.state = "playing"  # or "unknown"
+        source.supported_cmds = self.supported_cmds
```

Comment:
should the supported_cmds stuff be above line 147, so that even if we fail to munge the rest of the data we can still control the stream?